File: java/revenj-core/src/main/java/org/revenj/extensibility/SystemState.java
package org.revenj.extensibility;

public interface SystemState {
	boolean isBooting();

	boolean isReady();

	rx.Observable<Container> ready();

	rx.Observable<SystemEvent> change();

	void notify(SystemEvent value);

	class SystemEvent {

		public final String id;
		public final String detail;

		public SystemEvent(String id, String detail) {
			this.id = id;
			this.detail = detail;
		}
	}
}

File: scala/revenj-core/src/main/scala/net/revenj/extensibility/SystemState.scala
package net.revenj.extensibility

import monix.reactive.Observable

trait SystemState {
  def isBooting: Boolean

  def isReady: Boolean

  def ready: Observable[Container]

  def change: Observable[SystemState.SystemEvent]

  def notify(value: SystemState.SystemEvent): Unit
}

object SystemState {

  case class SystemEvent(id: String, detail: String)

}
